### PR TITLE
[tempo-distributed] Add helper for PodDisruptionBudget API Version

### PR DESCRIPTION
**Pull request by andy-mcgrath**
- Add helper tempo.pdb.apiVersion
- Update all PodDisruptionBudgetresourcesapiVersionto usetempo.pdb.apiVersion`

Fixes issue #1782
Signed-off-by: Andy McGrath <upon-pave0d@icloud.com>

### CODE REVIEW

**Comment by BitProcessor** [comment]
Hi @andy-mcgrath - thanks for your contribution. Seems like the docs need some fixing. 
See: https://github.com/grafana/helm-charts/runs/8227053520?check_suite_focus=true

File: charts/tempo-distributed/templates/_helpers.tpl [hunk 1]
```diff
@@ -120,3 +120,14 @@ Return if ingress supports pathType.
 {{- define "tempo.ingress.supportsPathType" -}}
   {{- or (eq (include "tempo.ingress.isStable" .) "true") (and (eq (include "tempo.ingress.apiVersion" .) "networking.k8s.io/v1beta1") (semverCompare ">= 1.18-0" .Capabilities.KubeVersion.Version)) -}}
 {{- end -}}
+
+{{/*
+Return the appropriate apiVersion for PodDisruptionBudget.
+*/}}
+{{- define "tempo.pdb.apiVersion" -}}
+  {{- if and (.Capabilities.APIVersions.Has "policy/v1") (semverCompare ">=1.21-0" .Capabilities.KubeVersion.Version) -}}
```
Comment (zanhsieh) [inline charts/tempo-distributed/templates/_helpers.tpl:128]: Actually there are two common fixes for this:

1. Stick on what Kubernetes version, which is how loki / loki-simple-distributed stick on.
```suggestion
  {{- if semverCompare ">=1.21-0" .Capabilities.KubeVersion.Version -}}
```

2. Stick on which API available, which is how grafana and prometheus related charts stick on.
```suggestion
  {{- if .Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" -}}
```

**Review by zanhsieh — COMMENTED** [review]
See my comment.

File: charts/tempo-distributed/templates/_helpers.tpl (hunk 1 above)
Comment (andy-mcgrath) [reply charts/tempo-distributed/templates/_helpers.tpl:128]: What's the preference? Personally I would have just check the API version but happy to update either way.

**Review by zanhsieh — APPROVED** [review]
See my comment.

**Pull request merged**